### PR TITLE
docs: add signature provider explanation

### PR DESCRIPTION
**Pull request by vrolland**

### CODE REVIEW

File: packages/docs/docs/guides/5-request-client/6-signature-provider.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
+---
+title: Use your own signature mechanism
+keywords: [Request, test network, signature, signatureProvider]
```
Comment (yomarion) [inline packages/docs/docs/guides/5-request-client/6-signature-provider.md:3]:
```suggestion
keywords: [Request, signature, signatureProvider]
```
I think test network is not relevant here

File: packages/docs/docs/guides/5-request-client/6-signature-provider.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  supportedMethods: Signature.METHOD[];
+  supportedIdentityTypes: Identity.TYPE[];
+
+  sign: (data: any, signer: Identity.IIdentity) => Promise<Signature.ISignedData>;
+}
+```
+
+## Example 1
+
+You have your own package to make signatures:
```
Comment (benjlevesque) [inline packages/docs/docs/guides/5-request-client/6-signature-provider.md:21]:
```suggestion
You have your own package to sign:
```

File: packages/docs/docs/guides/5-request-client/6-signature-provider.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+// We can initialize the RequestNetwork class with the signature provider
+const requestNetwork = new RequestNetwork.RequestNetwork({
+  mySignatureProvider,
+});
+```
+
+## Example 2
+
+You have your own package to make signatures:
```
Comment (benjlevesque) [inline packages/docs/docs/guides/5-request-client/6-signature-provider.md:101]:
```suggestion
You have your own package to sign:
```

File: packages/docs/docs/guides/5-request-client/6-signature-provider.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+   */
+  public async sign(
+    data: any,
+    signer: IdentityTypes.IIdentity,
+  ): Promise<SignatureTypes.ISignedData> {
+    if (!this.supportedIdentityTypes.includes(signer.type)) {
+      throw Error(`Identity type not supported ${signer.type}`);
+    }
+
+    const hashData = Utils.crypto.normalizeKeccak256Hash(data).value;
```
Comment (benjlevesque) [inline packages/docs/docs/guides/5-request-client/6-signature-provider.md:70]: could maybe use a comment here :)

File: packages/docs/docs/guides/5-request-client/6-signature-provider.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+Now you can inject it to the request client:
+
+```typescript
+import MySignatureProvider from 'mySignatureProvider';
+
+const mySignatureProvider = new MySignatureProvider();
+
+// We can initialize the RequestNetwork class with the signature provider
+const requestNetwork = new RequestNetwork.RequestNetwork({
+  mySignatureProvider,
```
Comment (benjlevesque) [inline packages/docs/docs/guides/5-request-client/6-signature-provider.md:95]:
```suggestion
  signatureProvider:  mySignatureProvider,
```

File: packages/docs/docs/guides/5-request-client/6-signature-provider.md (hunk 3 above)
Comment (benjlevesque) [reply packages/docs/docs/guides/5-request-client/6-signature-provider.md:101]: Maybe the introduction of example 2 shouldn't be the same as example 1?

File: packages/docs/docs/guides/5-request-client/6-signature-provider.md [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+ * Implementation of the signature provider for my wallet
+ */
+export default class MySignatureProvider
+  implements SignatureProviderTypes.ISignatureProvider {
+  /** list of supported signing method */
+  public supportedMethods: SignatureTypes.METHOD[] = [SignatureTypes.METHOD.ECDSA];
+  /** list of supported identity types */
+  public supportedIdentityTypes: IdentityTypes.TYPE[] = [IdentityTypes.TYPE.ETHEREUM_ADDRESS];
+
+  /** Dictionary containing all the private key indexed by address */
```
Comment (benjlevesque) [inline packages/docs/docs/guides/5-request-client/6-signature-provider.md:138]:
```suggestion
  /** Dictionary containing all the private keys indexed by address */
```

File: packages/docs/docs/guides/5-request-client/6-signature-provider.md [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+Now you can inject it to the request client:
+
+```typescript
+import MySignatureProvider from 'mySignatureProvider';
+
+const mySignatureProvider = new MySignatureProvider(anIdentity, aWalletId);
+
+// We can initialize the RequestNetwork class with the signature provider
+const requestNetwork = new RequestNetwork.RequestNetwork({
+  mySignatureProvider,
```
Comment (benjlevesque) [inline packages/docs/docs/guides/5-request-client/6-signature-provider.md:217]:
```suggestion
  signatureProvider: mySignatureProvider,
```

File: packages/docs/docs/guides/5-request-client/6-signature-provider.md [hunk 8]
```diff
@@ -0,0 +1,222 @@
+---
+title: Use your own signature mechanism
+keywords: [Request, test network, signature, signatureProvider]
+description: Learn how to integrate Request network and its features.
+---
+
+In a previous chapter, we used the signature provider `@requestnetwork/web3-signature`. But, if you are not using web3, you need to inject your own signature mechanism to the request client.
```
Comment (rittme) [inline packages/docs/docs/guides/5-request-client/6-signature-provider.md:7]: We also (mostly) used `@requestnetwork/epk-signature` on the previous chapters

File: packages/docs/docs/guides/5-request-client/6-signature-provider.md [hunk 9]
```diff
@@ -0,0 +1,222 @@
+---
+title: Use your own signature mechanism
+keywords: [Request, test network, signature, signatureProvider]
+description: Learn how to integrate Request network and its features.
```
Comment (rittme) [inline packages/docs/docs/guides/5-request-client/6-signature-provider.md:4]:
```suggestion
description: Learn how to build your own signature provider.
```

File: packages/docs/docs/guides/5-request-client/6-signature-provider.md [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+      signature: {
+        method: SignatureTypes.METHOD.ECDSA,
+        value: signatureValue,
+      },
+    };
+  }
+}
+```
+
+Now you can inject it to the request client:
```
Comment (rittme) [inline packages/docs/docs/guides/5-request-client/6-signature-provider.md:86]:
```suggestion
Now you can inject it into the request client:
```

File: packages/docs/docs/guides/5-request-client/6-signature-provider.md [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+    if (!this.supportedIdentityTypes.includes(identity.type)) {
+      throw Error(`Identity type not supported ${identity.type}`);
+    }
+
+    this.walletIdDictionary.set(identity.value.toLowerCase(), walletId);
+  }
+}
+```
+
+Now you can inject it to the request client:
```
Comment (rittme) [inline packages/docs/docs/guides/5-request-client/6-signature-provider.md:208]:
```suggestion
Now you can inject it into the request client:
```

File: packages/docs/docs/guides/5-request-client/6-signature-provider.md [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+import MySignatureProvider from 'mySignatureProvider';
+
+const mySignatureProvider = new MySignatureProvider(anIdentity, aWalletId);
+
+// We can initialize the RequestNetwork class with the signature provider
+const requestNetwork = new RequestNetwork.RequestNetwork({
+  mySignatureProvider,
+});
+
+// later on, you can even add more identity supported
```
Comment (rittme) [inline packages/docs/docs/guides/5-request-client/6-signature-provider.md:220]:
```suggestion
// later on, you can even add more supported identities
```

File: packages/docs/docs/guides/5-request-client/6-signature-provider.md [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+    if (identity && walletId) {
+      this.addSignatureParameters(identity, walletId);
+    }
+  }
+
+  /**
+   * Signs data
+   *
+   * @param string data the data to sign
+   * @returns IIdentity the identity to sign with if not given, the default signer will be used
```
Comment (ltacker) [inline packages/docs/docs/guides/5-request-client/6-signature-provider.md:153]:
```suggestion
   * @returns IIdentity the identity to sign with. If not given, the default signer will be used
```

File: packages/docs/docs/guides/5-request-client/6-signature-provider.md [hunk 14]
```diff
@@ -0,0 +1,224 @@
...
+
+// We can initialize the RequestNetwork class with the signature provider
+const requestNetwork = new RequestNetwork.RequestNetwork({
+  signatureProvider: mySignatureProvider,
+});
+```
+
+## Example 2
+
+You have your own package to sign:
```
Comment (rittme) [inline packages/docs/docs/guides/5-request-client/6-signature-provider.md:102]: This line should have more precisions about the example. It makes it look like they are both the same example.

**Pull request merged**